components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -489,6 +514,9 @@ public Response apisApiIdGraphqlSchemaPut(String apiId, String schemaDefinition,
     @Override
     public Response apisApiIdPut(String apiId, APIDTO body, String ifMatch, MessageContext messageContext) {
         APIDTO updatedApiDTO;
+        Object tokenInfo =
+                PhaseInterceptorChain.getCurrentMessage().getExchange().get(RestApiConstants.MESSAGE_EXCHANGE_TOKEN_INFO);
+        String tokenScopes[] = ((AccessTokenInfo) tokenInfo).getScopes();
```

Comment:
Let's use the existing constant.
USER_REST_API_SCOPES

Reply:
Also, let's handle the situation when the object is not available before casting and proceeding. We might probably return with a `403 Forbidden` when it is null.

Reply:
> Also, let's handle the situation when the object is not available before casting and proceeding. We might probably return with a `403 Forbidden` when it is null.

Fixed with bac9808b0af1a420ecf1fcc5b73c47e1783cdc74